shell/platform/android/surface/android_surface_transaction.h

```diff
@@ -0,0 +1,55 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_ANDROID_ANDROID_SURFACE_TRANSACTION_H_
+#define FLUTTER_SHELL_PLATFORM_ANDROID_ANDROID_SURFACE_TRANSACTION_H_
+
+#include "flutter/fml/macros.h"
+
+#include <cstdint>
+
+namespace flutter {
+
+//------------------------------------------------------------------------------
+/// The Android SurfaceTransaction API is used by `AndroidSurface` to
+/// synchronize frame presentation times. It's only available on API 29+.
+///
+class AndroidSurfaceTransaction {
+ public:
+  static AndroidSurfaceTransaction& GetInstance() {
```

Comment:
Move this out of the header file and into the source file.

Reply:
Done